### PR TITLE
WIP Fixes #224 - use custom openjdk12 container for circle

### LINKED ISSUE #224: Circle CI JDK12 support
Circle CI currently does not support JDK12. Latest Elastic requires JDK12. 

The main contributors are building this locally on a diverse set of local machines to CI before cutting releases, but we need to get our CI operational again once JDK12 container is released. 

We think this is important, so we'll keep the badge up on the README with the note why the failure is happening.

### PR DESCRIPTION
**Pull request by softwaredoug**
Please see this branch which currently doesn't build to attempt to use the provided jdk12 container. Currently stuck as it runs the commands as root, and Elasticsearch gives the error for each test

```
com.o19s.es.explore.ExplorerQueryBuilderTests > classMethod FAILED
    java.lang.RuntimeException: can not run elasticsearch as root
```

To recreate install circle tools locally and run:

```
circleci local execute --job build
```

### CODE REVIEW

**Comment by softwaredoug** [comment]
Working on a custom container instead of relying on this, probably just want to depend on our own container moving forward given how fast Elasticsearch moves with dependencies

File: .circleci/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    curl -LfsSo /tmp/openjdk.tar.gz ${JAVA_URL}; \
+    echo "${JAVA_SHA256} */tmp/openjdk.tar.gz" | sha256sum -c -; \
+    mkdir -p /opt/java/openjdk; \
+    cd /opt/java/openjdk; \
+    tar -xf /tmp/openjdk.tar.gz --strip-components=1; \
+    apk del --purge .fetch-deps; \
+    rm -rf /var/cache/apk/*; \
+    rm -rf /tmp/openjdk.tar.gz;
+
+RUN chmod a+rwx /opt/java/openjdk/bin/*
```
Comment (nomoa) [inline .circleci/Dockerfile]: nit: this gives all users the write permissions
if `a+rx` works then I think it'd be a bit safer

Comment (softwaredoug) [reply .circleci/Dockerfile]: thanks! will update

**Comment by softwaredoug** [comment]
we're still getting error 137 periodically, which seems due to memory issues spinning up the ES cluster for testing. Circle only gives us 4 GB of RAM...

**Comment by softwaredoug** [comment]
Changing back to WIP, doesn't seem like this error 137 is really fixed

**Comment by softwaredoug** [comment]
Closed in favor of #231 to switch to travis

**Pull request closed without merging**